src/app/gene-plot/gene-plot.component.spec.ts

```diff
@@ -131,4 +132,149 @@ describe('GenePlotComponent', () => {
     const filteredCircleTitlesOther = circleTitles.filter(item => item.textContent === alleleTitleOther);
     expect(filteredRectTitlesOther).toStrictEqual(filteredCircleTitlesOther);
   });
+
+  it('should create plot with different values', () => {
+    const summaryAllelesArray = new SummaryAllelesArray(variantsArrayMock);
+    Object.defineProperty(component, 'variantsArray', {
+      value: summaryAllelesArray
+    });
+    component.ngOnChanges({ gene: new SimpleChange(null, geneMock, true) });
+    component.redraw();
+    const plot = fixture.nativeElement as HTMLElement;
+    [
+      Array.from(plot.querySelectorAll('g polygon')),
+      Array.from(plot.querySelectorAll('g rect')),
+      Array.from(plot.querySelectorAll('svg g path'))
+    ].forEach((variantType, i) => {
+      expect(variantType).toHaveLength([1, 25, 5][i]);
+    });
+    Object.defineProperty(component, 'variantsArray', {
+      value: new SummaryAllelesArray([])
+    });
+    component.ngOnChanges({ gene: new SimpleChange(null, geneMock, true) });
+    component.redraw();
+    [
+      Array.from(plot.querySelectorAll('g polygon')),
+      Array.from(plot.querySelectorAll('g rect')),
+      Array.from(plot.querySelectorAll('svg g path'))
+    ].forEach((variantType, i) => {
+      expect(variantType).toHaveLength([0, 17, 4][i]);
+    });
+  });
+
+  it('should check viewbox and dimensions', () => {
+    const summaryAllelesArray = new SummaryAllelesArray(variantsArrayMock);
+    Object.defineProperty(component, 'variantsArray', {
+      value: summaryAllelesArray
+    });
+    component.ngOnChanges({ gene: new SimpleChange(null, geneMock, true) });
+    component.redraw();
+    expect(d3.selectAll('svg g rect').attr('height')).toBe('22');
+    expect(d3.selectAll('svg g rect').attr('width')).toBe('1835');
+    expect(d3.selectAll('#svg-container svg').attr('viewBox')).toBe('0 0 2000 517');
+  });
+
+  it('should call draw transcript on single chromosome', () => {
+    const spyOnDrawTranscript = jest.spyOn(component, 'drawTranscript');
+    const summaryAllelesArray = new SummaryAllelesArray(variantsArrayMock);
+    Object.defineProperty(component, 'variantsArray', {
+      value: summaryAllelesArray
+    });
+    component.ngOnChanges({ gene: new SimpleChange(null, geneMock, true) });
+    component.redraw();
+    expect(spyOnDrawTranscript).toHaveBeenCalledTimes(4);
+    [new Transcript('collapsed', 'chr3', '-', [
+      {
+        chromosome: 'chr3',
+        start: 120628380,
+        stop: 120682111
+      }
+    ], [
+      {
+        chromosome: 'chr3',
+        start: 120628168,
+        stop: 120628529
+      },
+      {
+        chromosome: 'chr3',
+        start: 120633147,
+        stop: 120633328
+      }
+    ])].forEach((trans, i) => {
```

Comment:
Why do we need an array with a forEach in this unit test, considering the array has just one element?